ProjFS.Mac/PrjFSKextTests/MockProc.cpp

```diff
@@ -149,6 +151,11 @@ int proc_selfpid(void)
     return s_selfPid;
 }
 
+void proc_selfname(char* buf, int size)
+{
+    strlcpy(buf, s_selfName.c_str(), size);
```

Comment:
Do we need to check the return value from `strlcpy` or ensure we were able to copy the full name into the provided buffer?

Reply:
We haven't been doing that in the code base.  I don't think it's a necessity for this test code.  It would certainly double check the behavior but for strlcpy but I think the behavior should be well defined.  Open to others onions on this too!

Reply:
cool - thanks for answering. I was just curious.

Reply:
In this specific case we're actually matching the behavior of the OS:

```
void
proc_selfname(char * buf, int  size)
{
	proc_t p;

	if ((p = current_proc())!= (proc_t)0) {
		strlcpy(buf, &p->p_comm[0], size);
	}
}
```
https://github.com/apple/darwin-xnu/blob/a449c6a3b8014d9406c2ddbdc81795da24aa7443/bsd/kern/kern_proc.c#L952

Reply:
The procname is always limited to `MAXCOMLEN` `char`s anyway, so any calling code will expect names to be truncated, as this is extremely common in practice.